src/uct/ib/ud/base/ud_ep.c

```diff
@@ -198,6 +207,12 @@ static UCS_CLASS_CLEANUP_FUNC(uct_ud_ep_t)
 
     ucs_trace_func("ep=%p id=%d conn_id=%d", self, self->ep_id, self->conn_id);
 
+    if (1 == iface->async.cbq_elem_on) {
+        uct_worker_slowpath_progress_unregister(iface->super.super.worker,
```

Comment:
unit common code - pls extract this block to a function and use it from uct_ud_iface_async_handler_remove() as well